test/PlaidClientTest.js

```
@@ -547,51 +548,107 @@ describe('plaid.Client', () => {
-                await_results: true
-              }
-            },
-            (err, mfaResponse, successResponse) => {
-              expect(err).to.be(null);
-              expect(successResponse).to.be.ok();
+      describe('transactions', () => {
+        let accessToken;
+        beforeEach(done => {
+          pCl.createItem({
+            username: testConstants.USERNAME,
+            password: testConstants.PASSWORDS.GOOD
+          }, testConstants.INSTITUTION, testConstants.PRODUCTS, {
+            transactions: {
+              start_date: now,
+              end_date: now,
+              await_results: true
+            }
+          },
+          (err, mfaResponse, successResponse) => {
+            expect(err).to.be(null);
+            expect(successResponse).to.be.ok();
+
+            accessToken = successResponse.access_token;
+            done();
+          });
+        });
+
+        it('normal flow', cb => {
+          pCl.getTransactions(accessToken, now, now, {},
+          (err, successResponse) => {
+            expect(err).to.be(null);
+            expect(successResponse).to.be.ok();
+            expect(successResponse.transactions).to.be.an(Array);
+
+            cb();
+          });
+        });
+
+        it('all transactions', cb => {
+          pCl.getAllTransactions(accessToken, now, now,
+          (err, transactions) => {
+            expect(err).to.be(null);
+            expect(transactions).to.be.an(Array);
+
+            cb();
+          });
+        });
 
-              cb(null, successResponse.access_token);
+        it('all < 500 transactions with correct pagination', cb => {
+          sinon.stub(pCl, 'getTransactions').callsFake(
+            (access_token, start_date, end_date, options) => {
+              if (options.offset === 0) {
+                return Promise.resolve({
+                  transactions: R.range(0, 200),
+                  total_transactions: 200,
+                });
+              }
```

i think we should throw an error in any other case